py3 flag for Vim 7.4.2119 fails

Hi there,

I need my Vim versions to be compiled with Python 3 and Vim 7.4 fails.

Here is my docker part:

```
FROM tweekmonster/vim-testbed:latest

RUN install_vim \
      -tag v7.4.2119 -py3 -build \
      -tag v8.1.2000 -py3 -build \
      -tag neovim:v0.2.0 -py3 -build \
      -tag neovim:v0.3.8 -py3 -build
```

and here is the output:

```
Step 1/11 : FROM tweekmonster/vim-testbed:latest
 ---> ce6ea355eebb
Step 2/11 : RUN install_vim       -tag v7.4.2119 -py3 -build       -tag v8.1.2000 -py3 -build       -tag neovim:v0.2.0 -py3 -build       -tag neovim:v0.3.8 -py3 -build
 ---> Running in 89800ec4f3bb
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/main/x86_64/APKINDEX.tar.gz
+ APK_BUILD_DEPS=
+ apk update
fetch http://dl-cdn.alpinelinux.org/alpine/v3.10/community/x86_64/APKINDEX.tar.gz
v3.10.3-19-g7f993019c4 [http://dl-cdn.alpinelinux.org/alpine/v3.10/main]
v3.10.3-13-g8068beb776 [http://dl-cdn.alpinelinux.org/alpine/v3.10/community]
OK: 10338 distinct packages available
=== building: NAME=, TAG=v7.4.2119, PYTHON=3, RUBY=0, LUA=0, FLAVOR= ===
+ init_vars
+ FLAVOR=
+ TAG=
+ NAME=
+ PYTHON2=
+ PYTHON3=
+ RUBY=0
+ LUA=0
+ CONFIGURE_OPTIONS=
+ PREBUILD_SCRIPT=
+ clean=
+ '[' 16 -gt 0 ]
+ TAG=v7.4.2119
+ shift
+ shift
+ '[' 14 -gt 0 ]
+ PYTHON3=3
+ shift
+ '[' 13 -gt 0 ]
+ echo '=== building: NAME=, TAG=v7.4.2119, PYTHON=3, RUBY=0, LUA=0, FLAVOR= ==='
+ prepare_build
+ '[' -z v7.4.2119 ]
+ IFS=: read -r -- repo tag
+ '[' -z  ]
+ tag=v7.4.2119
+ repo=
+ '[' -z  ]
+ FLAVOR=vim
+ '[' -z  ]
+ '[' vim '=' vim ]
+ repo=vim/vim
+ '[' -z ]
+ NAME=vim-v7.4.2119
+ '[' vim '=' vim ]
+ VIM_NAME=vim/vim/v7.4.2119_py3_rb0_lua0
+ INSTALL_PREFIX=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0
+ '[' vim '=' vim ]
+ VIM_CONFIG_ARGS='--prefix=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0 --enable-multibyte --without-x --enable-gui=no --with-compiledby=vim-testbed --with-tlib=ncurses'
+ set +x
+ apk_add_build_dep curl gcc libc-dev make
+ apk info -e curl
TAG:v7.4.2119
repo:vim/vim
tag:v7.4.2119
FLAVOR:vim
NAME:vim-v7.4.2119
+ apk add curl
(1/4) Installing ca-certificates (20190108-r0)
(2/4) Installing nghttp2-libs (1.39.2-r0)
(3/4) Installing libcurl (7.66.0-r0)
(4/4) Installing curl (7.66.0-r0)
Executing busybox-1.30.1-r2.trigger
Executing ca-certificates-20190108-r0.trigger
OK: 7 MiB in 18 packages
+ APK_BUILD_DEPS=' curl'
+ apk info -e gcc
+ apk add gcc
(1/10) Installing binutils (2.32-r0)
(2/10) Installing gmp (6.1.2-r1)
(3/10) Installing isl (0.18-r0)
(4/10) Installing libgomp (8.3.0-r0)
(5/10) Installing libatomic (8.3.0-r0)
(6/10) Installing libgcc (8.3.0-r0)
(7/10) Installing mpfr3 (3.1.5-r1)
(8/10) Installing mpc1 (1.1.0-r0)
(9/10) Installing libstdc++ (8.3.0-r0)
(10/10) Installing gcc (8.3.0-r0)
Executing busybox-1.30.1-r2.trigger
OK: 93 MiB in 28 packages
+ APK_BUILD_DEPS=' curl gcc'
+ apk info -e libc-dev
+ apk add libc-dev
(1/2) Installing musl-dev (1.1.22-r3)
(2/2) Installing libc-dev (0.7.1-r0)
OK: 104 MiB in 30 packages
+ APK_BUILD_DEPS=' curl gcc libc-dev'
+ apk info -e make
+ apk add make
(1/1) Installing make (4.2.1-r2)
Executing busybox-1.30.1-r2.trigger
OK: 104 MiB in 31 packages
+ APK_BUILD_DEPS=' curl gcc libc-dev make'
+ '[' -n  ]
+ '[' -n 3 ]
+ apk_add_build_dep python3-dev
+ apk info -e python3-dev
+ apk add python3-dev
(1/13) Installing pkgconf (1.6.1-r1)
(2/13) Installing libbz2 (1.0.6-r7)
(3/13) Installing expat (2.2.8-r0)
(4/13) Installing libffi (3.2.1-r6)
(5/13) Installing gdbm (1.13-r1)
(6/13) Installing xz-libs (5.2.4-r0)
(7/13) Installing ncurses-terminfo-base (6.1_p20190518-r0)
(8/13) Installing ncurses-terminfo (6.1_p20190518-r0)
(9/13) Installing ncurses-libs (6.1_p20190518-r0)
(10/13) Installing readline (8.0.0-r0)
(11/13) Installing sqlite-libs (3.28.0-r1)
(12/13) Installing python3 (3.7.5-r1)
(13/13) Installing python3-dev (3.7.5-r1)
Executing busybox-1.30.1-r2.trigger
OK: 213 MiB in 44 packages
+ APK_BUILD_DEPS=' curl gcc libc-dev make python3-dev'
+ '[' vim '=' vim ]
+ VIM_CONFIG_ARGS='--prefix=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0 --enable-multibyte --without-x --enable-gui=no --with-compiledby=vim-testbed --with-tlib=ncurses --enable-python3interp=dynamic'
+ '[' 0 -eq 1 ]
+ '[' 0 -eq 1 ]
+ '[' vim '=' vim ]
+ '[' -n  ]
+ cd /vim
+ '[' -d /vim-build/vim/vim/v7.4.2119_py3_rb0_lua0 ]
+ BUILD_DIR=vim-vim/vim-v7.4.2119
+ '[' '!' -d vim-vim/vim-v7.4.2119 ]
+ mkdir -p vim-vim/vim-v7.4.2119
+ cd vim-vim/vim-v7.4.2119
+ url=https://github.com/vim/vim/archive/v7.4.2119.tar.gz
+ echo 'Downloading vim/vim:v7.4.2119 from https://github.com/vim/vim/archive/v7.4.2119.tar.gz'
Downloading vim/vim:v7.4.2119 from https://github.com/vim/vim/archive/v7.4.2119.tar.gz
+ tar zx '--strip-components=1'
+ curl --retry 3 -SL https://github.com/vim/vim/archive/v7.4.2119.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   118    0   118    0     0     33      0 --:--:--  0:00:03 --:--:--    33
100 12.2M    0 12.2M    0     0   8483      0 --:--:--  0:25:14 --:--:-- 10741
+ '[' vim '=' vim ]
+ apk_add_build_dep ncurses-dev
+ apk info -e ncurses-dev
+ apk add ncurses-dev
(1/1) Installing ncurses-dev (6.1_p20190518-r0)
Executing busybox-1.30.1-r2.trigger
OK: 213 MiB in 45 packages
+ APK_BUILD_DEPS=' curl gcc libc-dev make python3-dev ncurses-dev'
+ apk add ncurses
(1/1) Installing ncurses (6.1_p20190518-r0)
Executing busybox-1.30.1-r2.trigger
OK: 214 MiB in 46 packages
+ build
+ '[' -n  ]
+ '[' vim '=' vim ]
+ sed -n '/^MAJOR = / s~MAJOR = ~~p' Makefile
+ MAJOR=7
+ '[' 7 -lt 8 ]
+ sed -i 's~sys/time.h termio.h~sys/time.h sys/types.h termio.h~' src/configure.in src/auto/configure
Configuring with: --prefix=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0 --enable-multibyte --without-x --enable-gui=no --with-compiledby=vim-testbed --with-tlib=ncurses --enable-python3interp=dynamic
+ echo 'Configuring with: --prefix=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0 --enable-multibyte --without-x --enable-gui=no --with-compiledby=vim-testbed --with-tlib=ncurses --enable-python3interp=dynamic'
+ ./configure '--prefix=/vim-build/vim/vim/v7.4.2119_py3_rb0_lua0' --enable-multibyte --without-x '--enable-gui=no' '--with-compiledby=vim-testbed' '--with-tlib=ncurses' '--enable-python3interp=dynamic'
configure: creating cache auto/config.cache
checking whether make sets $(MAKE)... yes
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables...
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for fgrep... /bin/grep -F
checking for library containing strerror... none required
checking for gawk... no
checking for mawk... no
checking for nawk... no
checking for awk... awk
checking for strip... strip
checking for ANSI C header files... yes
checking for sys/wait.h that is POSIX.1 compatible... yes
checking --enable-fail-if-missing argument... no
checking for recent clang version... no
configure: checking for buggy tools...
checking for BeOS... no
checking for QNX... no
checking for Darwin (Mac OS X)... no
checking AvailabilityMacros.h usability... no
checking AvailabilityMacros.h presence... no
checking for AvailabilityMacros.h... no
checking --with-local-dir argument... Defaulting to /usr/local
checking --with-vim-name argument... Defaulting to vim
checking --with-ex-name argument... Defaulting to ex
checking --with-view-name argument... Defaulting to view
checking --with-global-runtime argument... no
checking --with-modified-by argument... no
checking if character set is EBCDIC... no
checking --disable-smack argument... no
checking linux/xattr.h usability... no
checking linux/xattr.h presence... no
checking for linux/xattr.h... no
checking --disable-selinux argument... no
checking for is_selinux_enabled in -lselinux... no
checking --with-features argument... Defaulting to huge
checking --with-compiledby argument... vim-testbed
checking --disable-xsmp argument... no
checking --disable-xsmp-interact argument... no
checking --enable-luainterp argument... no
checking --enable-mzschemeinterp argument... no
checking --enable-perlinterp argument... no
checking --enable-pythoninterp argument... no
checking --enable-python3interp argument... dynamic
checking for python3... /usr/bin/python3
checking Python version... 3.7
checking Python is 3.0 or better... yep
checking Python's abiflags... m
checking Python's install prefix... /usr
checking Python's execution prefix... /usr
checking Python's configuration directory... /usr/lib/python3.7/config-3.7m-x86_64-linux-gnu
checking Python3's dll name... libpython3.7m.so.1.0
checking if -pthread should be used... yes
checking if compile and link flags for Python 3 are sane... yes
checking --enable-tclinterp argument... no
checking --enable-rubyinterp argument... no
checking --enable-cscope argument... no
checking --enable-workshop argument... no
checking --disable-netbeans argument... no
checking --disable-channel argument... no
checking for socket in -lsocket... no
checking for gethostbyname in -lnsl... no
checking whether compiling with process communication is possible... yes
checking --enable-multibyte argument... yes
checking --enable-hangulinput argument... no
checking --enable-xim argument... defaulting to auto
checking --enable-fontset argument... no
defaulting to: don't HAVE_X11
checking --enable-gui argument... no GUI support
checking X11/SM/SMlib.h usability... no
checking X11/SM/SMlib.h presence... no
checking for X11/SM/SMlib.h... no
checking for CYGWIN or MSYS environment... no
checking whether toupper is broken... no
checking whether __DATE__ and __TIME__ work... yes
checking whether __attribute__((unused)) is allowed... yes
checking elf.h usability... yes
checking elf.h presence... yes
checking for elf.h... yes
checking for main in -lelf... no
checking for dirent.h that defines DIR... yes
checking for library containing opendir... none required
checking stdint.h usability... yes
checking stdint.h presence... yes
checking for stdint.h... yes
checking stdlib.h usability... yes
checking stdlib.h presence... yes
checking for stdlib.h... yes
checking string.h usability... yes
checking string.h presence... yes
checking for string.h... yes
checking sys/select.h usability... yes
checking sys/select.h presence... yes
checking for sys/select.h... yes
checking sys/utsname.h usability... yes
checking sys/utsname.h presence... yes
checking for sys/utsname.h... yes
checking termcap.h usability... yes
checking termcap.h presence... yes
checking for termcap.h... yes
checking fcntl.h usability... yes
checking fcntl.h presence... yes
checking for fcntl.h... yes
checking sgtty.h usability... no
checking sgtty.h presence... no
checking for sgtty.h... no
checking sys/ioctl.h usability... yes
checking sys/ioctl.h presence... yes
checking for sys/ioctl.h... yes
checking sys/time.h usability... yes
checking sys/time.h presence... yes
checking for sys/time.h... yes
checking sys/types.h usability... yes
checking sys/types.h presence... yes
checking for sys/types.h... yes
checking termio.h usability... no
checking termio.h presence... no
checking for termio.h... no
checking iconv.h usability... yes
checking iconv.h presence... yes
checking for iconv.h... yes
checking inttypes.h usability... yes
checking inttypes.h presence... yes
checking for inttypes.h... yes
checking langinfo.h usability... yes
checking langinfo.h presence... yes
checking for langinfo.h... yes
checking math.h usability... yes
checking math.h presence... yes
checking for math.h... yes
checking unistd.h usability... yes
checking unistd.h presence... yes
checking for unistd.h... yes
checking stropts.h usability... yes
checking stropts.h presence... yes
checking for stropts.h... yes
checking errno.h usability... yes
checking errno.h presence... yes
checking for errno.h... yes
checking sys/resource.h usability... yes
checking sys/resource.h presence... yes
checking for sys/resource.h... yes
checking sys/systeminfo.h usability... no
checking sys/systeminfo.h presence... no
checking for sys/systeminfo.h... no
checking locale.h usability... yes
checking locale.h presence... yes
checking for locale.h... yes
checking sys/stream.h usability... no
checking sys/stream.h presence... no
checking for sys/stream.h... no
checking termios.h usability... yes
checking termios.h presence... yes
checking for termios.h... yes
checking libc.h usability... no
checking libc.h presence... no
checking for libc.h... no
checking sys/statfs.h usability... yes
checking sys/statfs.h presence... yes
checking for sys/statfs.h... yes
checking poll.h usability... yes
checking poll.h presence... yes
checking for poll.h... yes
checking sys/poll.h usability... yes
checking sys/poll.h presence... yes
checking for sys/poll.h... yes
checking pwd.h usability... yes
checking pwd.h presence... yes
checking for pwd.h... yes
checking utime.h usability... yes
checking utime.h presence... yes
checking for utime.h... yes
checking sys/param.h usability... yes
checking sys/param.h presence... yes
checking for sys/param.h... yes
checking libintl.h usability... no
checking libintl.h presence... no
checking for libintl.h... no
checking libgen.h usability... yes
checking libgen.h presence... yes
checking for libgen.h... yes
checking util/debug.h usability... no
checking util/debug.h presence... no
checking for util/debug.h... no
checking util/msg18n.h usability... no
checking util/msg18n.h presence... no
checking for util/msg18n.h... no
checking frame.h usability... no
checking frame.h presence... no
checking for frame.h... no
checking sys/acl.h usability... no
checking sys/acl.h presence... no
checking for sys/acl.h... no
checking sys/access.h usability... no
checking sys/access.h presence... no
checking for sys/access.h... no
checking sys/sysinfo.h usability... yes
checking sys/sysinfo.h presence... yes
checking for sys/sysinfo.h... yes
checking wchar.h usability... yes
checking wchar.h presence... yes
checking for wchar.h... yes
checking wctype.h usability... yes
checking wctype.h presence... yes
checking for wctype.h... yes
checking for sys/ptem.h... no
checking for sys/sysctl.h... no
checking for pthread_np.h... no
checking strings.h usability... yes
checking strings.h presence... yes
checking for strings.h... yes
checking if strings.h can be included after string.h... yes
checking whether gcc needs -traditional... no
checking for an ANSI C-conforming const... yes
checking for working volatile... yes
checking for mode_t... yes
checking for off_t... yes
checking for pid_t... yes
checking for size_t... yes
checking for uid_t in sys/types.h... yes
checking for uint32_t... yes
checking whether time.h and sys/time.h may both be included... yes
checking for ino_t... yes
checking for dev_t... yes
checking whether byte ordering is bigendian... no
checking for inline... inline
checking for rlim_t... yes
checking for stack_t... yes
checking whether stack_t has an ss_base field... no
checking --with-tlib argument... ncurses
checking for linking with ncurses library... OK
checking for tgetent()... yes
checking whether we talk terminfo... yes
checking whether termcap.h contains ospeed... yes
checking whether termcap.h contains UP, BC and PC... yes
checking whether tputs() uses outfuntype... no
checking whether sys/select.h and sys/time.h may both be included... yes
checking for /dev/ptc... no
checking for SVR4 ptys... yes
checking for ptyranges... don't know
checking default tty permissions/group... can't determine - assume ptys are world accessible
world
checking return type of signal handlers... void
checking for struct sigcontext... yes
checking getcwd implementation is broken... no
checking for bcmp... yes
checking for fchdir... yes
checking for fchown... yes
checking for fsync... yes
checking for getcwd... yes
checking for getpseudotty... no
checking for getpwent... yes
checking for getpwnam... yes
checking for getpwuid... yes
checking for getrlimit... yes
checking for gettimeofday... yes
checking for getwd... no
checking for lstat... yes
checking for memcmp... yes
checking for memset... yes
checking for mkdtemp... yes
checking for nanosleep... yes
checking for opendir... yes
checking for putenv... yes
checking for qsort... yes
checking for readlink... yes
checking for select... yes
checking for setenv... yes
checking for setpgid... yes
checking for setsid... yes
checking for sigaltstack... yes
checking for sigstack... no
checking for sigset... yes
checking for sigsetjmp... yes
checking for sigaction... yes
checking for sigvec... no
checking for strcasecmp... yes
checking for strerror... yes
checking for strftime... yes
checking for stricmp... no
checking for strncasecmp... yes
checking for strnicmp... no
checking for strpbrk... yes
checking for strtol... yes
checking for tgetent... yes
checking for towlower... yes
checking for towupper... yes
checking for iswupper... yes
checking for usleep... yes
checking for utime... yes
checking for utimes... yes
checking for _LARGEFILE_SOURCE value needed for large files... no
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... no
checking for st_blksize... yes
checking whether stat() ignores a trailing slash... no
checking for iconv_open()... yes
checking for nl_langinfo(CODESET)... yes
checking for strtod in -lm... yes
checking for strtod() and other floating point functions... yes
checking for isinf()... yes
checking for isnan()... yes
checking --disable-acl argument... no
checking for acl_get_file in -lposix1e... no
checking for acl_get_file in -lacl... no
checking for POSIX ACL support... no
checking for acl_get in -lsec... no
checking for Solaris ACL support... no
checking for AIX ACL support... no
checking --disable-gpm argument... no
checking for gpm... no
checking --disable-sysmouse argument... no
checking for sysmouse... no
checking for FD_CLOEXEC... yes
checking for rename... yes
checking for sysctl... not usable
checking for sysinfo... yes
checking for sysinfo.mem_unit... yes
checking for sysconf... yes
checking size of int... 4
checking size of long... 8
checking size of time_t... 8
checking size of off_t... 8
checking uint32_t is 32 bits... ok
checking whether memmove handles overlaps... yes
checking for _xpg4_setrunelocale in -lxpg4... no
checking how to create tags... ctags
checking how to run man with a section nr... man
checking --disable-nls argument... no
checking for msgfmt... no
checking for NLS... msgfmt not found - disabled
checking dlfcn.h usability... yes
checking dlfcn.h presence... yes
checking for dlfcn.h... yes
checking for dlopen()... yes
checking for dlsym()... yes
checking setjmp.h usability... yes
checking setjmp.h presence... yes
checking for setjmp.h... yes
checking for GCC 3 or later... yes
checking whether we need -D_FORTIFY_SOURCE=1... yes
checking linker --as-needed support... yes
configure: updating cache auto/config.cache
configure: creating auto/config.status
config.status: creating auto/config.mk
config.status: creating auto/config.h
+ make 'CFLAGS=-U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2' -j4
Starting make in the src directory.
If there are problems, cd to the src directory and run make there
cd src && make first
make[1]: Entering directory '/vim/vim-vim/vim-v7.4.2119/src'
mkdir objects
CC="gcc -Iproto -DHAVE_CONFIG_H       " srcdir=. sh ./osdef.sh
cd xxd; CC="gcc" CFLAGS=" -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2" LDFLAGS="-L/usr/local/lib -Wl,--as-needed" \
	make -f Makefile
creating auto/pathdef.c
make[2]: Entering directory '/vim/vim-vim/vim-v7.4.2119/src/xxd'
gcc -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2 -L/usr/local/lib -Wl,--as-needed -DUNIX -o xxd xxd.c
make[2]: Leaving directory '/vim/vim-vim/vim-v7.4.2119/src/xxd'
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/arabic.o arabic.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/buffer.o buffer.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/blowfish.o blowfish.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/charset.o charset.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/crypt.o crypt.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/crypt_zip.o crypt_zip.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/dict.o dict.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/diff.o diff.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/digraph.o digraph.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/edit.o edit.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/eval.o eval.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/evalfunc.o evalfunc.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ex_cmds.o ex_cmds.c
ex_cmds.c: In function 'fix_help_buffer':
ex_cmds.c:6900:12: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     if (e1 == NUL || e2 == NUL)
            ^~
ex_cmds.c:6900:9: note: did you mean to dereference the pointer?
     if (e1 == NUL || e2 == NUL)
         ^
ex_cmds.c:6900:25: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     if (e1 == NUL || e2 == NUL)
                         ^~
ex_cmds.c:6900:22: note: did you mean to dereference the pointer?
     if (e1 == NUL || e2 == NUL)
                      ^
ex_cmds.c: In function 'set_context_in_sign_cmd':
ex_cmds.c:8192:11: warning: comparison between pointer and zero character constant [-Wpointer-compare]
     if (p == NUL)
           ^~
ex_cmds.c:8192:9: note: did you mean to dereference the pointer?
     if (p == NUL)
         ^
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ex_cmds2.o ex_cmds2.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ex_docmd.o ex_docmd.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ex_eval.o ex_eval.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ex_getln.o ex_getln.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/farsi.o farsi.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/fileio.o fileio.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/fold.o fold.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/getchar.o getchar.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/hardcopy.o hardcopy.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/hashtab.o hashtab.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/if_cscope.o if_cscope.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/if_xcmdsrv.o if_xcmdsrv.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/list.o list.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/mark.o mark.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/memline.o memline.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/menu.o menu.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/misc1.o misc1.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/misc2.o misc2.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/move.o move.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/mbyte.o mbyte.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/normal.o normal.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ops.o ops.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2          -I/usr/include/python3.7m -DPYTHON3_HOME='L"/usr"' -pthread -DDYNAMIC_PYTHON3_DLL=\"libpython3.7m.so.1.0\"   -o objects/option.o option.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/os_unix.o os_unix.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/pathdef.o auto/pathdef.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/popupmnu.o popupmnu.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/quickfix.o quickfix.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/regexp.o regexp.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/screen.o screen.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/search.o search.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/sha256.o sha256.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/spell.o spell.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/spellfile.o spellfile.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/syntax.o syntax.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/tag.o tag.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/term.o term.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/ui.o ui.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/undo.o undo.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/userfunc.o userfunc.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/window.o window.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -I/usr/include/python3.7m -DPYTHON3_HOME='L"/usr"' -pthread -DDYNAMIC_PYTHON3_DLL=\"libpython3.7m.so.1.0\"  -o objects/if_python3.o if_python3.c
In file included from /usr/include/python3.7m/pystate.h:11,
                 from /usr/include/python3.7m/traceback.h:8,
                 from /usr/include/python3.7m/Python.h:119,
                 from if_python3.c:75:
/usr/include/python3.7m/pythread.h:122:5: error: #error "Require native threads. See https://bugs.python.org/issue31370"
 #   error "Require native threads. See https://bugs.python.org/issue31370"
     ^~~~~
/usr/include/python3.7m/pythread.h:131:5: error: unknown type name 'NATIVE_TSS_KEY_T'
     NATIVE_TSS_KEY_T _key;
     ^~~~~~~~~~~~~~~~
if_python3.c:179: warning: "PySlice_GetIndicesEx" redefined
 # define PySlice_GetIndicesEx py3_PySlice_GetIndicesEx

In file included from /usr/include/python3.7m/Python.h:120,
                 from if_python3.c:75:
/usr/include/python3.7m/sliceobject.h:48: note: this is the location of the previous definition
 #define PySlice_GetIndicesEx(slice, length, start, stop, step, slicelen) (  \

gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/netbeans.o netbeans.c
gcc -c -I. -Iproto -DHAVE_CONFIG_H     -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2       -o objects/channel.o channel.c
make[1]: *** [Makefile:2964: objects/if_python3.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory '/vim/vim-vim/vim-v7.4.2119/src'
make: *** [Makefile:29: first] Error 2
+ bail 'Make failed'
+ echo 'Make failed'
Make failed
+ exit 1
The command '/bin/sh -c install_vim       -tag v7.4.2119 -py3 -build       -tag v8.1.2000 -py3 -build       -tag neovim:v0.2.0 -py3 -build       -tag neovim:v0.3.8 -py3 -build' returned a non-zero code: 1
```

--- a/example/Dockerfile
+++ b/example/Dockerfile
@@ -5,7 +5,7 @@ RUN apk add make
 
 RUN install_vim -tag v7.1 -name vim71 -prebuild_script 'echo "#define FEAT_PROFILE" >> src/feature.h' -build \
                 -tag v7.3.429 -name vim73 -py -build \
-                -tag v7.4.052 -name vim74-trusty -build \
+                -tag v7.4.052 -name vim74-trusty -py3 -build \
                 -tag master -py2 -py3 -ruby -lua -build \
                 -tag neovim:v0.2.0 -py2 -py3 -ruby -build \
                 -tag neovim:master -py2 -py3 -ruby -build

--- a/scripts/install_vim.sh
+++ b/scripts/install_vim.sh
@@ -246,6 +246,11 @@ build() {
       DEPS_CMAKE_FLAGS="$DEPS_CMAKE_FLAGS -DUSE_BUNDLED_UNIBILIUM=ON"
     fi
 
+    if grep -qF 'UTF8PROC' CMakeLists.txt; then
+      apk add utf8proc
+      apk_add_build_dep utf8proc-dev
+    fi
+
     # NOTE: uses "make cmake" to avoid linking twice when changing versiondef.h
     make cmake CMAKE_BUILD_TYPE=RelWithDebInfo \
       CMAKE_EXTRA_FLAGS="$CMAKE_EXTRA_FLAGS" \

--- a/scripts/install_vim.sh
+++ b/scripts/install_vim.sh
@@ -207,6 +207,9 @@ build() {
       sed -i 's~sys/time.h termio.h~sys/time.h sys/types.h termio.h~' src/configure.in src/auto/configure
     fi
 
+    # Apply Vim patch v8.0.1635 to fix build with Python.
+    sed -i '/#ifdef _POSIX_THREADS/,+2 d' src/if_python3.c
+
     echo "Configuring with: $VIM_CONFIG_ARGS"
     # shellcheck disable=SC2086
     ./configure $VIM_CONFIG_ARGS || bail "Could not configure"

--- a/scripts/install_vim.sh
+++ b/scripts/install_vim.sh
@@ -200,15 +200,17 @@ build() {
   if [ "$FLAVOR" = vim ]; then
     # Apply build fix from v7.1.148.
     # NOTE: this silently does nothing with 7.1.148+, but can be skipped with
-    # Vim 8+ (and needs to for 8.0.0082, where src/configure.in was renamed to
-    # src/configure.ac).
+    # Vim 8+ (and needs to be for 8.0.0082, where src/configure.in was renamed
+    # to src/configure.ac).
     MAJOR="$(sed -n '/^MAJOR = / s~MAJOR = ~~p' Makefile)"
     if [ "$MAJOR" -lt 8 ]; then
       sed -i 's~sys/time.h termio.h~sys/time.h sys/types.h termio.h~' src/configure.in src/auto/configure
     fi
 
     # Apply Vim patch v8.0.1635 to fix build with Python.
-    sed -i '/#ifdef _POSIX_THREADS/,+2 d' src/if_python3.c
+    if grep -q _POSIX_THREADS src/if_python3.c; then
+      sed -i '/#ifdef _POSIX_THREADS/,+2 d' src/if_python3.c
+    fi
 
     echo "Configuring with: $VIM_CONFIG_ARGS"
     # shellcheck disable=SC2086

